presto/presto.go

```diff
@@ -1366,67 +1351,7 @@ func getScanType(typeNames []string) (reflect.Type, error) {
 		v = sql.NullInt64{}
 	case "real", "double":
 		v = sql.NullFloat64{}
-	case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-		v = sql.NullTime{}
-	case "map":
-		v = NullMap{}
-	case "array":
-		if len(typeNames) <= 1 {
-			return nil, ErrInvalidResponseType
-		}
-		switch typeNames[1] {
-		case "boolean":
-			v = NullSliceBool{}
-		case "json", "char", "varchar", "varbinary", "interval year to month", "interval day to second", "decimal", "ipaddress", "uuid", "unknown":
-			v = NullSliceString{}
-		case "tinyint", "smallint", "integer", "bigint":
-			v = NullSliceInt64{}
-		case "real", "double":
-			v = NullSliceFloat64{}
-		case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-			v = NullSliceTime{}
-		case "map":
-			v = NullSliceMap{}
-		case "array":
-			if len(typeNames) <= 2 {
-				return nil, ErrInvalidResponseType
-			}
-			switch typeNames[2] {
-			case "boolean":
-				v = NullSlice2Bool{}
-			case "json", "char", "varchar", "varbinary", "interval year to month", "interval day to second", "decimal", "ipaddress", "uuid", "unknown":
-				v = NullSlice2String{}
-			case "tinyint", "smallint", "integer", "bigint":
-				v = NullSlice2Int64{}
-			case "real", "double":
-				v = NullSlice2Float64{}
-			case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-				v = NullSlice2Time{}
-			case "map":
-				v = NullSlice2Map{}
-			case "array":
-				if len(typeNames) <= 3 {
-					return nil, ErrInvalidResponseType
-				}
-				switch typeNames[3] {
-				case "boolean":
-					v = NullSlice3Bool{}
-				case "json", "char", "varchar", "varbinary", "interval year to month", "interval day to second", "decimal", "ipaddress", "uuid", "unknown":
-					v = NullSlice3String{}
-				case "tinyint", "smallint", "integer", "bigint":
-					v = NullSlice3Int64{}
-				case "real", "double":
-					v = NullSlice3Float64{}
-				case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
-					v = NullSlice3Time{}
-				case "map":
-					v = NullSlice3Map{}
-				}
-				// if this is a 4 or more dimensional array, scan type will be an empty interface
-			}
-		}
-	}
```

Comment:
This array stuff was way more complicated than we need it to be, for our purposes. It was trying to parse the arrays into properly typed Go slices, but we just need to display it back to the user in the format in which it's returned from Presto. So I removed this code and am now just returning arrays (as well as maps and rows) as strings.